src/PoolManager.sol

```diff
@@ -442,31 +440,21 @@ contract PoolManager is Auth {
         return pools[poolId].tranches[trancheId].liquidityPools[currencyIdToAddress[currencyId]];
     }
 
-    function getLiquidityPool(uint64 poolId, bytes16 trancheId, address currencyAddress)
-        public
-        view
-        returns (address)
-    {
-        return pools[poolId].tranches[trancheId].liquidityPools[currencyAddress];
+    function getLiquidityPool(uint64 poolId, bytes16 trancheId, address currency) public view returns (address) {
+        return pools[poolId].tranches[trancheId].liquidityPools[currency];
     }
 
-    function getTrancheTokenPrice(uint64 poolId, bytes16 trancheId, address currencyAddress)
+    function getTrancheTokenPrice(uint64 poolId, bytes16 trancheId, address currency)
         public
         view
         returns (uint256 price, uint64 computedAt)
     {
-        TrancheTokenPrice memory value = pools[poolId].tranches[trancheId].prices[currencyAddress];
+        TrancheTokenPrice memory value = pools[poolId].tranches[trancheId].prices[currency];
         price = value.price;
         computedAt = value.computedAt;
     }
 
-    function isAllowedAsInvestmentCurrency(uint64 poolId, address currencyAddress) public view returns (bool) {
-        uint128 currency = currencyAddressToId[currencyAddress];
-        if (currency == 0) {
```

Comment:
This is already checked in `addCurrency`. And `allowInvestmentCurrency` can only be called after `addCurrency`.